### DISCUSSION TITLE
Login with GitHub

### DISCUSSION THREAD

**Discussion by bondz** [npm/feedback#75] [General]
It would be fantastic if the website supported GitHub as a login mechanism. The service is owned by GitHub, but I was wondering if there were any plans already in progress or in the future to implement this. It's one less login to remember.

This could allow for npm GitHub package registry packages to also be listed on npm.
2FA is also already setup or required for some organizations on GitHub and that could allow users access an organization's registry on npm.

**Comment by MylesBorins** [comment]
Hey @bondz this is a fantastic idea and something we definitely have talked about around the office (virtually of course). For this quarter we wanted to focus on small improvements to make the general npm experience better, but I can assure you that login with GitHub is on our radar. I’ll talk with the team and see about timelines and if we think it makes sense to add to the roadmap just yet.

**Comment by arcanis** [comment]
Duplicate with https://github.com/npm/rfcs/discussions/93

**Reply by MylesBorins** [reply]
I don't see it as a duplicate but rather as a complementary feature. Please feel free to correct me if I'm wrong, but I'm seeing GitHub auth as a combination of the way that it is done (browser based login is one possibility), and what gets unlocked based on the connecting of accounts (such as making tokens available in GitHub actions).

With that in mind I don't think this makes sense to mark this as a dupe, but rather as related.

**Reply by arcanis** [reply]
Aaaah - I misread the "if the website" part 😃 I thought this was about making the "Connect via GitHub" workflow available for the login CLI - which would be pretty nice.

**Reply by MylesBorins** [reply]
FWIW I am imagining this is likely a broader account linking / auth with GitHub initiative which would likely be a combo of work in CLI / Website / Registry / GitHub
